### PR TITLE
MRG, ENH: Make fixed length epochs

### LINKED ISSUE #4367: ENH: change _segment_raw to make_fixed_length_epochs
```
from mne.datasets import testing
import os.path as op
from mne.io import read_raw_fif
from mne import Epochs, make_fixed_length_events
from numpy.testing import assert_equal

elekta_base_dir = op.join(testing.data_path(download=False), 'misc')
fname_raw_elekta = op.join(elekta_base_dir, 'test_elekta_3ch_raw.fif')

def test_shape():
    raw = read_raw_fif(fname_raw_elekta, preload=True)
    sfreq = raw.info['sfreq']
    events = make_fixed_length_events(raw, id=1, duration=20.)
    epochs = Epochs(raw, events, tmin=0., tmax=20.)
    epochs_ok = Epochs(raw, events, tmin=0, tmax=20. - 1./sfreq)
    data = epochs.get_data()
    data_ok = epochs_ok.get_data()
    assert_equal(data.shape, (4, 4, 20000))
    assert_equal(data_ok.shape, (4, 4, 20000))

if __name__ == '__main__':

    test_shape()
```

The following test fails because of the first `assert_equal`. 
Here, the `raw` object contains data with 4 channels and 80 000 samples (sfreq = 1000Hz).
I want to epoch the raw data and have epochs which are all 20s long. For that, I used the `make_fixed_length_events` function. Intuitively, I expected the shape of the epoched data to be (4, 4, 20000). However, it is actually (3, 4, 20001). It seems that there is one redundant sample. This issue (if it is really one), is easily corrected by replacing `tmax=20.` with `tmax=20. - 1./sfreq` in `Epochs`. 

Is this happening because I made a mistake somewhere ?

### PR DESCRIPTION
**Pull request by mdaneshzand**
#### Reference issue
Example: Fixes #4367 

#### What does this implement/fix?
fixed some of the bugs

#### Additional information
Any additional information you think is important.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -3080,14 +3081,14 @@ def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
         Length of each segment in seconds. Defaults to 1.
     verbose: bool
         Whether to report what is being done by printing text.
-    **kwargs
-        Any additional keyword arguments are passed to ``Epochs`` constructor.
-
+    preload: bool
```
Comment (larsoner) [inline mne/epochs.py]:
```suggestion
    %(preload)s
    %(verbose)s
```
Can use auto-fillers (`%(something)s`) here and also maintain the same order in this list that there is above

Also please add a section below `Returns` like:
```
    Notes
    -----
    .. versionadded:: 0.20

File: mne/epochs.py [hunk 2]
```diff
@@ -3069,25 +3069,30 @@ def average_movements(epochs, head_pos=None, orig_sfreq=None, picks=None,
...
-        Length of each segment in seconds. Defaults to 1.
+    duration : float
+        Duration of each epoch in seconds. Defaults to 1.
     verbose: bool
         Whether to report what is being done by printing text.
-    **kwargs
-        Any additional keyword arguments are passed to ``Epochs`` constructor.
-
+    %(preload)s
+    %(verbose)s
```
Comment (larsoner) [inline mne/epochs.py]: Need newline between this and `Returns` to get Sphinx to render properly

File: mne/epochs.py [hunk 3]
```diff
@@ -3069,25 +3069,30 @@ def average_movements(epochs, head_pos=None, orig_sfreq=None, picks=None,
...
     Returns
+    Notes
```
Comment (larsoner) [inline mne/epochs.py]: Need newline between, and also `Notes` should go after `Returns` like:
```
Returns
-------
epochs : ...
    ...

Notes
-----
.. versionadded:: 0.20
```

File: mne/epochs.py [hunk 4]
```diff
@@ -3069,25 +3069,32 @@ def average_movements(epochs, head_pos=None, orig_sfreq=None, picks=None,
 
 
 @verbose
-def _segment_raw(raw, segment_length=1., verbose=None, **kwargs):
+def make_fixed_length_epochs(raw, duration=1.,
+                             preload=False, verbose=None):
```
Comment (jasmainak) [inline mne/epochs.py]: you need to update `python_reference.rst`

**Comment by larsoner** [comment]
Thanks @mdaneshzand

**Comment by mmagnuski** [comment]
👍

**Pull request merged**